test/Sentry.Tests/HubTests.cs

```diff
@@ -66,13 +66,13 @@ public void PushAndLockScope_DoesNotAffectOuterScope()
         var hub = _fixture.GetSut();
 
         // Act & assert
-        hub.ConfigureScope(s => Assert.False(s.Locked));
+        hub.ScopeManager.ConfigureScope(s => Assert.False(s.Locked));
         using (hub.PushAndLockScope())
         {
-            hub.ConfigureScope(s => Assert.True(s.Locked));
+            hub.ScopeManager.ConfigureScope(s => Assert.True(s.Locked));
         }
 
-        hub.ConfigureScope(s => Assert.False(s.Locked));
+        hub.ScopeManager.ConfigureScope(s => Assert.False(s.Locked));
```

Comment:
test: This test fails during CI (`Run Android API-34 Test (net9.0)` and `Run Android API-36 Test (net9.0)`)

Previously, assertions were using `hub.ConfigureScope(scope => ...));`.
But `Hub.ConfigureScope(Action<Scope>)` catches all exceptions.
https://github.com/getsentry/sentry-dotnet/blob/bc91dd3bf6cf844e239ac429892d95b9ce00a21b/src/Sentry/Internal/Hub.cs#L96-L106

Switching to instead use `Hub.ScopeManager.ConfigureScope(Action<Scope>)` no longer catches any exceptions in the callback:
https://github.com/getsentry/sentry-dotnet/blob/6e50eaa87b3aa24a921824c4494566b01f0640f0/src/Sentry/Internal/SentryScopeManager.cs#L35-L39

In short:
We were previously hiding a failed assertion that is now surfaced.

Reply:
Since this is a P0 and it's unrelated to this PR, we should probably create a new issue for this and deal with it separately.

Reply:
I have updated the test to reflect and assert this expectation.